src/service-override/storage.ts

```diff
@@ -130,17 +131,25 @@ class ExternalStorageService extends AbstractStorageService {
 }
 
 registerServiceInitializePreParticipant(async (accessor) => {
-  await (accessor.get(IStorageService) as ExternalStorageService).initialize()
+  const storageService = accessor.get(IStorageService)
+  const hostService = accessor.get(IHostService)
+  if (storageService instanceof AbstractStorageService) {
+    await storageService.initialize()
+
+    hostService.onDidChangeFocus(focus => {
+      if (!focus) {
+        void storageService.flush()
```

Comment:
Isn't this normally done by VSCode? Or is it treeshaken away?

Reply:
same: done a in non-included code: https://github.com/microsoft/vscode/blob/dd9e1d579c6f97205f35fcbe73fae49b0051e375/src/vs/workbench/browser/workbench.ts#L255-L259